File: Leetcode/typescript/sol1752.ts

// https://leetcode.com/problems/check-if-array-is-sorted-and-rotated/

function check(nums: number[]): boolean {
    let count:number = 0;
    for(let i = 0; i < nums.length; i++){
        if(nums[i] > nums[(i+1)%nums.length]){ //  check that at mosth there is once contiguos values are decreasing
            count++;
        }
        if(count > 1){
            return false;
        }
    }
    return true;
};

File: Leetcode/python/sol1752.py
#
#
# class Solution:
#     def check(self, nums: List[int]) -> bool:
#         for i in range(len(nums)):
#             prev = i
#             isGood = True
#             for _ in range(len(nums)-1):
#                 if nums[prev] > nums[(prev+1)%len(nums)]:
#                     isGood = False
#                     break
#                 prev = (prev+1)%len(nums)
#             if isGood:
#                 return True
#         return False


x = 12
print(x&1)
x<<=1
print(x)
x<<=1
print(x)
